python/cudf/cudf/core/join/join.py

```diff
@@ -354,7 +377,9 @@ def _sort_result(self, result: Frame) -> Frame:
                 sort_order = result._get_sorted_inds(
                     list(_coerce_to_tuple(self.on))
                 )
-            return result._gather(sort_order, keep_index=False)
+            return result._gather(
+                sort_order, keep_index=False, check_bounds=False
```

Comment:
I wonder if there are other places that we might leverage this optimization (probably out of scope for this PR, just want to plant the idea in your head).

Reply:
There's definitely a bunch.